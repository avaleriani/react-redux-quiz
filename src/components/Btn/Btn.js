import React from "react";
import styles from "./styles.module.css";

/**
 * @types {{action: function, text: string, type: string}} props
 * @return HTMLElement
 */
const Btn = (props) => (
  <button onClick={props.action ? props.action : null} type={props.type ? props.type : "button"}
          className={styles['button']}>{props.text}</button>
);

export default Btn;