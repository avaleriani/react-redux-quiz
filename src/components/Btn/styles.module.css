.button {
    border: 2px solid #D0DB97;
    color: #D0DB97;
    padding: 10px 15px 10px 15px;
    background: transparent;
    transition: border-color 500ms ease, color 500ms ease;
    cursor: pointer;
    font-size: 1.25rem;
    width: 150px;
    margin: auto;
    outline: none;
}

.button:hover {
    color: #69B578;
    border: 2px solid #69B578;
}