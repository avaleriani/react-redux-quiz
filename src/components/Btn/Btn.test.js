import React from 'react';
import renderer from 'react-test-renderer';
import Btn from './Btn';

describe('Btn component', () => {

  test('renders correctly without props', () => {
    const btnComponent = renderer.create(<Btn/>).toJSON();
    expect(btnComponent).toMatchSnapshot();
  });

  test('renders with props', () => {
    const btnComponent = renderer.create(<Btn action={() => {
    }} type="button" text="text"/>).toJSON();
    expect(btnComponent).toMatchSnapshot();
  });
});