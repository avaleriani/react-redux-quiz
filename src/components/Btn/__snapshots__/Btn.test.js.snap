// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn component renders correctly without props 1`] = `
<button
  className="button"
  onClick={null}
  type="button"
/>
`;

exports[`Btn component renders with props 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  text
</button>
`;
